mkosi/sandbox.py

```diff
@@ -150,12 +155,11 @@ def sandbox_cmd(
     if network and not relaxed:
         cmdline += ["--bind", "/etc/resolv.conf", "/etc/resolv.conf"]
 
-    if devices:
-        shm = ":"
-    else:
-        shm = "chmod 1777 /dev/shm"
+    # bubblewrap creates everything with a restricted mode so relax stuff as needed.
+    shm = "chmod 1777 /dev/shm && " if not devices else ""
+    etc = "chmod 755 /etc && " if not relaxed else ""
 
-    cmdline += ["sh", "-c", f"{shm} && exec $0 \"$@\""]
+    cmdline += ["sh", "-c", f"{shm} {etc} exec $0 \"$@\"".strip()]
```

Comment:
Though it's a bit longer I think something like this might be a bit easier to read like a top down script
```python
script = []
if not devices:
    script += ["chmod 1777 /dev/shm"]
if not relaxed:
    script += ["chmod 755 /etc"]
script += ['exec $0 "$@"']
return ["sh", "-c", *[" && ".join(script)]]
```
Also doesn't need manual management for the && to be in the right place.